app/validators/placeholders/placeholder_validator.py

```diff
@@ -143,24 +143,32 @@ def validate_answer_and_transform_unit_match(self, *, arguments, transform_type)
         value = arguments["value"]
         answer_id = value.get("identifier")
         unit = arguments["unit"]
+        source_answer_ids = []
 
-        if self.errors or (
-            unit
-            == self.questionnaire_schema.answers_with_context[answer_id]["answer"][
-                "unit"
-            ]
-        ):
-            return None
+        if value.get("source") == "calculated_summary":
+            if calculated_summary_source_answer_ids := self.questionnaire_schema.get_calculation_block_ids(
+                block=self.questionnaire_schema.get_block(value.get("identifier")),
+                source_type="answers",
+            ):
+                source_answer_ids = calculated_summary_source_answer_ids
 
-        self.add_error(
-            error_messages.ANSWER_UNIT_AND_TRANSFORM_UNIT_MISMATCH.format(
-                answer_unit=self.questionnaire_schema.answers_with_context[answer_id][
-                    "answer"
-                ]["unit"],
-                transform_unit=unit,
-            ),
-            identifier=answer_id,
-        )
+        else:
+            source_answer_ids.append(value.get("identifier"))
+
+        for source_answer_id in source_answer_ids:
+
+            answers_unit = self.questionnaire_schema.answers_with_context[
+                source_answer_id
+            ]["answer"]["unit"]
+
+            if unit != answers_unit:
+                self.add_error(
+                    error_messages.ANSWER_UNIT_AND_TRANSFORM_UNIT_MISMATCH.format(
+                        answer_unit=answers_unit,
+                        transform_unit=unit,
+                    ),
+                    identifier=answer_id,
```

Comment:
as mentioned below, if `answer_id` was changed to identifier, then you could separately have `answer_id=source_answer_id` included in the error